src/core/services/IronBankService.ts

```diff
@@ -94,31 +94,48 @@ export class IronBankServiceImpl implements IronBankService {
     const { MAX_UINT256 } = this.config;
     const provider = this.web3Provider.getSigner();
     const ironBankMarketContract = getContract(marketAddress, ironBankMarketAbi, provider);
-
     switch (action) {
       case 'supply':
-        return await this.transactionService.execute({ network, fn: ironBankMarketContract.mint, args: [amount] });
+        return await this.transactionService.execute({
+          network,
```

Comment:
Network seems to be a user setting that is not changed based on every call right?

The IronBankService could have an internal variable network that gets updated through a setNetwork and that only gets called on the action that the user changes the network through the UI, this reduces the boilerplate of having to send it all the time.

Does marketAddress work in the same way or would marketAddress also change depending on which action the user is taking?